src/plugins/telemetry/server/telemetry_collection/get_data_telemetry/get_data_telemetry.ts

```diff
@@ -0,0 +1,244 @@
+      return false;
+    }
+    return new RegExp(`^${pattern.replace(/\./g, '\\.').replace(/\*/g, '.*')}$`).test(name);
+  });
+}
+
+function increaseCounters(
+  previousValue: DataTelemetryBasePayload = { index_count: 0 },
+  { isECS, docCount, sizeInBytes }: DataTelemetryIndex
+) {
+  return {
+    ...previousValue,
+    index_count: previousValue.index_count + 1,
+    ...(typeof isECS === 'boolean'
+      ? {
+          ecs_index_count: (previousValue.ecs_index_count || 0) + (isECS ? 1 : 0),
+        }
+      : {}),
+    ...(typeof docCount === 'number'
+      ? { doc_count: (previousValue.doc_count || 0) + docCount }
+      : {}),
+    ...(typeof sizeInBytes === 'number'
+      ? { size_in_bytes: (previousValue.size_in_bytes || 0) + sizeInBytes }
+      : {}),
+  };
+}
+
+export function buildDataTelemetryPayload(indices: DataTelemetryIndex[]): DataTelemetryPayload {
+  const startingDotPatternsUntilTheFirstAsterisk = DATA_DATASETS_INDEX_PATTERNS.map(({ pattern }) =>
+    pattern.replace(/^\.(.+)\*.*$/g, '.$1')
+  ).filter(Boolean);
+
+  // Filter out the system indices unless they are required by the patterns
+  const indexCandidates = indices.filter(
+    ({ name }) =>
+      !(
+        name.startsWith('.') &&
+        !startingDotPatternsUntilTheFirstAsterisk.find((pattern) => name.startsWith(pattern))
+      )
+  );
+
+  const acc = new Map<string, DataTelemetryDocument>();
+
+  for (const indexCandidate of indexCandidates) {
+    const matchingDescriptors = findMatchingDescriptors(indexCandidate);
+    if (matchingDescriptors) {
+      const { datasetName, datasetType, shipper, patternName } = matchingDescriptors;
+      const key = `${datasetName}-${datasetType}-${shipper}-${patternName}`;
+      acc.set(key, {
+        ...((datasetName || datasetType) && { dataset: { name: datasetName, type: datasetType } }),
+        ...(shipper && { shipper }),
+        ...(patternName && { pattern_name: patternName }),
+        ...increaseCounters(acc.get(key), indexCandidate),
+      });
+    }
+  }
+
+  return [...acc.values()];
+}
+
+interface IndexStats {
+  indices: {
+    [indexName: string]: {
+      total: {
+        docs: {
+          count: number;
+          deleted: number;
+        };
+        store: {
+          size_in_bytes: number;
+        };
+      };
+    };
+  };
+}
+
+interface IndexMappings {
+  [indexName: string]: {
+    mappings: {
+      _meta?: {
+        beat?: string;
+      };
+      properties: {
+        dataset?: {
+          properties: {
+            name?: {
+              type: string;
+              value?: string;
+            };
+            type?: {
+              type: string;
+              value?: string;
+            };
+          };
+        };
+        ecs?: {
+          properties: {
+            version?: {
+              type: string;
+            };
+          };
+        };
+      };
+    };
+  };
+}
+
+export async function getDataTelemetry(callCluster: APICaller) {
```

Comment:
We might want to create a task for this rather hefty request if it becomes an issue

Reply:
Yeah! I considered that approach. But since we don't need to paginate at all, I thought it might initially be OK to run it as is. Only applying `filter_path` for privacy and performance reasons (avoiding getting too much info) :)